### PR TITLE
Added fetchPrice example, reorganized HC docs

### PR DESCRIPTION
**Pull request by katieclay**
:clipboard: Add associated issues, tickets, docs URL here.

## Overview

Added a brief explanation of the `fetchPrice` contract

## Changes

Describe your changes and implementation choices. More details make PRs easier to review.

- Added new file for fetchPrice
- Reorganized HC docs to accommodate for both examples, including TOC tree

## Testing

n/a

### CODE REVIEW

**Comment by katieclay** [comment]
Need to ask about `server.register_function(offchain_addsub2, selector("offchain_addsub2(uint32,uint32)"))`; other example `selector()`s' parameter functions don't have "`offchain_`" in them.

File: .gitbook/contents/hc/1_price-fetch.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+```solidity
+    function fetchPrice(string calldata token) public {
+        string memory price;
+
+        // Encode function signature, which is called on the offchain rpc server.
+        bytes memory req = abi.encodeWithSignature("getprice(string)", token);
+        bytes32 userKey = bytes32(abi.encode(msg.sender));
+        (uint32 error, bytes memory ret) = HA.CallOffchain(userKey, req);
+
+        if (error != 0) {
```
Comment (mmontour1306) [inline .gitbook/contents/hc/1_price-fetch.md:45]: This is correct from a documentation perspective (matches the code) but I'm not sure if the code itself makes sense. Kevin, can you check this? I'm not sure why there's an "emit" followed by a "revert", or why it's splitting the error code and 'ret' string across two different events.

File: .gitbook/contents/hc/3_server-setup.md
Comment (mmontour1306) [inline .gitbook/contents/hc/3_server-setup.md:1]: It's not a change in this PR but I noticed a typo in this file. In the "why is that" section you show the JSON-RPC request sent to the server, including a method field:

  "method":"ASD97e0d7ba",
  
  That should be:
    "method":"97e0d7ba",

**Review by mmontour1306 — APPROVED** [review]
Made a couple of comments but it looks OK to me overall.

File: .gitbook/contents/hc/1_price-fetch.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+            emit FetchPriceError(error);
+            emit FetchPriceRet(ret);
```
Comment (wsdt) [inline .gitbook/contents/hc/1_price-fetch.md]:
```suggestion
       
```

agree

**Review by wsdt — APPROVED** [review]

**Pull request merged**